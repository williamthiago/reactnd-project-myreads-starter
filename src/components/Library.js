import React from 'react'
import { Link } from 'react-router-dom'

import Shelf from './Shelf'
import shelfType from '../enums/shelfType'

const Library = ({ books, loading }) => {
  const shelves = Object.keys(shelfType).map(type => {
    let name = shelfType[type]
    let booksInShelf = books.filter(book => book.shelf === type)
    return { type, name, booksInShelf }
  })
  
  return (
    <div className="list-books">
      <div className="list-books-title">
        <h1>MyReads</h1>
      </div>
      <div className="list-books-content">
        <div>
          {loading && 
            <p>Loading...</p>
          }
          {!loading && shelves.map(shelf => (
            <Shelf key={shelf.type} title={shelf.name} books={shelf.booksInShelf} />
          ))}
        </div>
      </div>
      <div className="open-search">
        <Link to="/search">Add a book</Link>
      </div>
    </div>
  )
}

export default Library
