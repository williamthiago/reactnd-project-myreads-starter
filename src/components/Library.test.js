import React from 'react'
import { shallow } from 'enzyme'

import Library from './Library'

it('Renders Library without books', () => {
  const books = []
  const onChangeShelf = () => {}
  expect(shallow(<Library books={books} onChangeShelf={onChangeShelf} />)).toMatchSnapshot()
})