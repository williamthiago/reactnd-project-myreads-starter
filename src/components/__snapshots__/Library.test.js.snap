// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Library without books 1`] = `
<div
  className="list-books"
>
  <div
    className="list-books-title"
  >
    <h1>
      MyReads
    </h1>
  </div>
  <div
    className="list-books-content"
  >
    <div>
      <Shelf
        books={Array []}
        onChangeShelf={[Function]}
        title="Currently Reading"
      />
      <Shelf
        books={Array []}
        onChangeShelf={[Function]}
        title="Want To Read"
      />
      <Shelf
        books={Array []}
        onChangeShelf={[Function]}
        title="Read"
      />
    </div>
  </div>
  <div
    className="open-search"
  >
    <Link
      replace={false}
      to="/search"
    >
      Add a book
    </Link>
  </div>
</div>
`;
